### ISSUE TITLE
PyTorch 2.6 mismatch

### ISSUE THREAD

**Issue by WangGosh** [PowerHouseMan/ComfyUI-AdvancedLivePortrait#89]
Weights only load failed. This file can still be loaded, to do so you have two options, [1mdo those steps only if you trust the source of the checkpoint[0m.
(1) In PyTorch 2.6, we changed the default value of the `weights_only` argument in `torch.load` from `False` to `True`. Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
(2) Alternatively, to load with `weights_only=True` please check the recommended steps in the following error message.
WeightsUnpickler error: Unsupported global: GLOBAL ultralytics.nn.tasks.DetectionModel was not an allowed global by default. Please use `torch.serialization.add_safe_globals([DetectionModel])` or the `torch.serialization.safe_globals([DetectionModel])` context manager to allowlist this global if you trust this class/function.

Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.

**Comment by devastrar** [comment]
Add this to the top of `custom_nodes/comfyui-advancedliveportrait/nodes.py` under `from ultralytics import YOLO`

```python
from ultralytics.nn.tasks import DetectionModel
from ultralytics.nn.modules.conv import Conv, Concat
from ultralytics.nn.modules.block import C2f, Bottleneck, SPPF, DFL
from ultralytics.nn.modules.head import Detect
from torch.nn import Sequential
from torch.nn.modules.conv import Conv2d
from torch.nn.modules.batchnorm import BatchNorm2d
from torch.nn.modules.activation import SiLU
from torch.nn.modules.container import ModuleList
from torch.nn.modules.pooling import MaxPool2d
from torch.nn.modules.upsampling import Upsample
torch.serialization.add_safe_globals([DetectionModel, Sequential, Conv, Conv2d, BatchNorm2d, SiLU, C2f, ModuleList, Bottleneck, SPPF, MaxPool2d, Upsample, Concat, Detect, DFL])
```

**Comment by WangGosh** [comment]
> Add this to the top of `custom_nodes/comfyui-advancedliveportrait/nodes.py` under `from ultralytics import YOLO`
> 
> from ultralytics.nn.tasks import DetectionModel
> from ultralytics.nn.modules.conv import Conv, Concat
> from ultralytics.nn.modules.block import C2f, Bottleneck, SPPF, DFL
> from ultralytics.nn.modules.head import Detect
> from torch.nn import Sequential
> from torch.nn.modules.conv import Conv2d
> from torch.nn.modules.batchnorm import BatchNorm2d
> from torch.nn.modules.activation import SiLU
> from torch.nn.modules.container import ModuleList
> from torch.nn.modules.pooling import MaxPool2d
> from torch.nn.modules.upsampling import Upsample
> torch.serialization.add_safe_globals([DetectionModel, Sequential, Conv, Conv2d, BatchNorm2d, SiLU, C2f, ModuleList, Bottleneck, SPPF, MaxPool2d, Upsample, Concat, Detect, DFL])

Thank you for your answer. I will try later when I have time.

**Comment by GitTonb** [comment]
This method can address this problem very well.

**Comment by MrCurio** [comment]
I got the same problem and after adding that info to the nodes.py file and reloading comfyUI, i still get the error. Any hints @GitTonb ?

Thanks

**Comment by devastrar** [comment]
> I got the same problem and after adding that info to the nodes.py file and reloading comfyUI, i still get the error. 

What is the exact error message? It will tell you what to add. The original error message was...

> GLOBAL ultralytics.nn.tasks.DetectionModel was not an allowed global by default. Please use torch.serialization.add_safe_globals([DetectionModel])

AdvancedLivePortrait is loading the yolo model in a way the violates the newer security. It won't run it unless the whatever the yolo model implements is included as a safe global. So if the error message is `GLOBAL xxxx.yyy.zzz.Something was not an allowed global by default. Please use torch.serialization.add_safe_globals([Something])`. Then add an import statement `from xxx.yyy.zzz import Something` and add `Something` to the `torch.serialization.add_safe_globals([Something1,Something2])` list. A different yolo model could implement different stuff.

**Comment by MrCurio** [comment]
I'm getting this error after adding those imports and safeglobals into the nodes.py file:

`!!! Exception during processing !!! Weights only load failed. This file can still be loaded, to do so you have two options, do those steps only if you trust the source of the checkpoint. 
        (1) In PyTorch 2.6, we changed the default value of the weights_only argument in torch.load from False to True. Re-running torch.load with weights_only set to False will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
        (2) Alternatively, to load with weights_only=True please check the recommended steps in the following error message.
        WeightsUnpickler error: Unsupported global: GLOBAL ultralytics.nn.tasks.DetectionModel was not an allowed global by default. Please use torch.serialization.add_safe_globals([DetectionModel]) or the torch.serialization.safe_globals([DetectionModel]) context manager to allowlist this global if you trust this class/function.`



EDIT: managed to make it working by adding the weights to False in motion_extractor.py file when loading the model. It is not recommended, but at least it works

here's the updated line:

`    def load_pretrained(self, init_path: str):
        if init_path not in (None, ''):

            state_dict = torch.load(init_path, map_location=lambda storage, loc: storage, weights_only=False)['model']   # << Here

            state_dict = filter_state_dict(state_dict, remove_name='head')
            ret = self.detector.load_state_dict(state_dict, strict=False)
            print(f'Load pretrained model from {init_path}, ret: {ret}')`

**Comment by Popshells** [comment]
I downloaded and installed this node to solve this issue instead: https://github.com/ltdrdata/comfyui-unsafe-torch

**Comment by liangzhengwei789** [comment]
> I downloaded and installed this node to solve this issue instead: https://github.com/ltdrdata/comfyui-unsafe-torch

This works, thank you
